.changeset/lazy-dancers-push.md

```diff
@@ -0,0 +1,7 @@
+---
+"openapi-typescript-helpers": patch
+"openapi-react-query": patch
+"openapi-fetch": patch
```

Comment:
I view this as a bug fix that implements the expected behavior. However, it does have the potential to break type checking in projects with incorrect code, so I can change this to `minor` if we want to be safe.